hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -4207,26 +4208,31 @@ protected Durability getEffectiveDurability(Durability d) {
   @Override
   public boolean checkAndMutate(byte[] row, byte[] family, byte[] qualifier, CompareOperator op,
     ByteArrayComparable comparator, TimeRange timeRange, Mutation mutation) throws IOException {
-    checkMutationType(mutation, row);
+    checkMutationType(mutation);
+    checkRow(mutation, row);
     return doCheckAndRowMutate(row, family, qualifier, op, comparator, null, timeRange, null,
```

Comment:
would it be better to put `checkMutationType()` and `checkRow()` in `doCheckAndRowMutate()` method?

Reply:
@virajjasani Thank you for reviewing this.

Sure. I'll do that.